File: com/bridgelab/dataStructure/CalendarWithQueue.java
package com.bridgelab.dataStructure;

import com.bridgeLab.dataStructure.calendar.Calendar;
import com.bridgelab.utility.Queue;

public class CalendarWithQueue {

	public Queue weekDayWithQueue(String[][] arr) {
		Queue queue = new Queue();
		for(int i=0; i<7; i++) {
			for(int j=0; j<7; j++) {
				queue.enqueue(arr[i][j]);
			}
		}
		return queue;
	}
	
	public void display(Queue queue) {
		int size = queue.size();
		for(int i=0; i<=size; i++) {
			if(i%7==0) System.out.println();
			String str = (String)queue.dequeue();
			if(str==null) break;
			System.out.print(str+"   ");
		}
	}
	
	public static void main(String args[]) {
		CalendarWithQueue calendar = new CalendarWithQueue();
		Calendar cal = new Calendar();
		String arr[][] = cal.calendar(2019, 1);
//		cal.display(arr);
		Queue queue = calendar.weekDayWithQueue(arr);
		calendar.display(queue);
	}
}

File: Month1/com/bridgelab/DataStructure/CalendarWithQueue.py
from com.bridgelab.DataStructure.Calendar import Calendar
from com.bridgelab.util.Queue import Queue
import numpy


class CalendarWithQueue:
    def __init__(self):
        self.queue = Queue()
        self.arr = numpy.array((6, 7))

    def calendar_with_queue(self, month, year):
        calendar = Calendar()
        arr = calendar.initial(month, year)
        arr1 = numpy.array(arr).reshape(6, 7)
        for i in arr1:
            self.queue.enqueue(i)
        for i in range(self.queue.size()):
            if i < 7:
                print(" ", end=" ")
            print(self.queue.dequeue())


calendar_with_queue = CalendarWithQueue()
month = int(input("Enter a month : "))
year = int(input("Enter a year : "))
calendar_with_queue.calendar_with_queue(month, year)
